ee/clickhouse/views/test/test_clickhouse_retention.py

```diff
@@ -337,6 +373,13 @@ class Breakdown:
     property: str
 
 
+class PropertyFilter(TypedDict):
```

Comment:
This type is incomplete - group_type_index is missing.

Ignoring the context of adding this completely - given this is a copy of posthog/models/property.py, can we unify the two somehow?

Reply:
I'm sure we can, although I prefer to keep the generation of the request payload as transparent as possible and not jump over the http interface tie to internal implementation. Advantage being the http interface is going to be more stable, where as the interface to `Property` less so.

Happy to change though if you think it's worth it.

Reply:
> Advantage being the http interface is going to be more stable, where as the interface to Property less so.

I think this is where our disagreement comes in - I think decoupling those two would be create more work with no payoff as there's an extra layer to debug whenever something goes wrong.